src/countsHelper.cpp

```diff
@@ -0,0 +1,31 @@
+#include <Rcpp.h>
+#include <unordered_map>
+#include <vector>
+using namespace Rcpp;
+
+//' @title
+//' Count number of occurences
+//'
+//' @param features a Matrix with doubles.
+//'
+//' @export
+// [[Rcpp::export]]
+NumericMatrix  countsToDhist(NumericMatrix features)
+{
+    std::unordered_map<double,int> counter;
```

Comment:
Can I double check my understanding of the `unordered_map` here? Does the syntax `map[val]` index into the map by it's key?

Reply:
Yep, see http://www.cplusplus.com/reference/unordered_map/unordered_map/operator[]/